pkg/cache/cache.go

```diff
@@ -0,0 +1,26 @@
+package cache
+
+import (
+	"github.com/miekg/dns"
+)
+
+//go:generate mockgen -destination=mock_$GOPACKAGE/$GOFILE . Cache
+
+type Cache interface {
+	Add(request, response *dns.Msg)
+	Get(request *dns.Msg) (response *dns.Msg)
+}
+
+func New(settings Settings) Cache {
+	settings.setDefaults()
+
+	switch settings.Type {
+	case LRU:
+		return newLRU(settings.MaxEntries, settings.TTL)
+	case NOOP:
+		return newNoop()
+	default:
+		// Coding error
+		panic("unknown cache type: " + settings.Type)
+	}
+}
```

Comment:
Maybe split into `NewNoop` and `NewLRU`? That way each could have their own settings too.